test_project/test_app/tests/test_storage.py

```diff
@@ -171,6 +175,37 @@ def test_store_and_remove_all_multi(self):
         keys = self.redis.keys('djac.test.metric*')
         self.assertEqual(len(keys), 0)
 
+    @mock.patch.object(uuid, 'uuid4')
+    def test_remove_intermediate_results_exact_suggest(self, mock_uuid):
+        """
+        After exact_suggest call, all intermediate result sets are removed
+        """
+        setattr(auto_settings, 'MAX_EXACT_MATCH_WORDS', 2)
+        autocomp = Autocompleter('stock')
+        autocomp.store_all()
+
+        mock_uuid.return_value = 'mock_uuid'
+        autocomp.exact_suggest('aapl')
+        mock_uuid.assert_called_once()
+        keys = self.redis.keys('djac.results.mock_uuid')
```

Comment:
to assume random term keys aren't left either, l would make this  a pattern
keys = self.redis.keys('djac.results.*) 
this way you are sure the per term results are cleared as well and as a bonus you don't need to do the mock_uuid stuff